### PR TITLE
Ansible playbook simplification around SGX drivers

### PR DESCRIPTION
**Pull request by jumaffre**
- It is now expected that drivers are included in the underlying Ubuntu distribution.
- The `sgx-psw-dev.yml` has been removed because it was unused. 
- Remove Open Enclave from `app_run` image
- Fix issue with CCF `deb_install.yml` playbook when using `latest` tag
- Remove CCF `install.yml` (from source) playbook as we always use `deb_install.yml` to install CCF 

TODO: 
- [x] Re-create fresh 20.04 VM 
- [x] Update docs

### CODE REVIEW

File: getting_started/setup_vm/roles/ccf_build/vars/common.yml [hunk 1]
```diff
@@ -25,8 +25,6 @@ debs:
   - iptables # partition test infra
   - libclang1-9 # required by doxygen
   - libclang-cpp9 # required by doxygen
-  - flex # required to build doxygen
```
Comment (achamayou) [inline getting_started/setup_vm/roles/ccf_build/vars/common.yml:28]: Good catch!

File: .azure-pipelines-gh-pages.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ trigger:
 
 jobs:
   - job: build_and_publish_docs
-    container: ccfciteam/ccf-ci:oe0.17.1-focal-clang10
+    container: ccfciteam/ccf-ci:oe0.17.1-focal-driverless
```
Comment (achamayou) [inline .azure-pipelines-gh-pages.yml:10]: We never installed the driver in containers, so this is a slightly odd name, but not a problem.

**Review by achamayou — APPROVED** [review]

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+- `app_run`: Builds the image containing all runtime dependencies for CCF, as well as the latest release of CCF (as per https://github.com/microsoft/CCF/releases/latest). To be used by CCF operators.
+- `app_ci`: Builds the image containing all build dependencies for CCF applications. To be used by CCF application developers.
+- `ccf_ci`: Builds the image containing all build dependencies for CCF itself. To be used by CCF contributors. It is also used by CCF Continuous Integration pipeline.
+
+To build a given image, run:
+
+```bash
+$ cd CCF/
+$ docker build -t <tag> -f docker/<app_run/app_ci/ccf_ci> .
```
Comment (eddyashton) [inline docker/README.md:11]:
```suggestion
$ docker build -t <tag> -f docker/<app_run|app_ci|ccf_ci> .
```

Comment (jumaffre) [reply docker/README.md:11]: Thanks! I knew something was odd when I jotted it down.

File: .azure-pipelines-gh-pages.yml (hunk 2 above)
Comment (jumaffre) [reply .azure-pipelines-gh-pages.yml:10]: Yes, you're right. I'll update the name of the image.

**Review by achamayou — APPROVED** [review]

**Pull request merged**